src/test/ui/span/issue-24690.stderr

```diff
@@ -4,6 +4,7 @@ error: unused variable: `theOtherTwo`
 20 |     let theOtherTwo = 2;
    |         ^^^^^^^^^^^
    |
+   = note: #[deny(warnings)] implies #[deny(unused_variables)]
```

Comment:
I agree that this is desirable, but doing it properly will involve addressing [the `FIXME`](https://github.com/rust-lang/rust/blob/164619a8cf/src/librustc/session/mod.rs#L322-L325) about generalizing one-time diagnostics beyond `span_note`, which looks sufficiently complicated that I'd prefer to do it in a separate PR.

(The most obvious thing to do would be to extract [the body of what is now](https://github.com/rust-lang/rust/blob/164619a8cf/src/librustc/session/mod.rs#L329-L342) `diag_span_note_once` into a more general `diag_once` that takes an enum argument to decide which method to call (`span_note` or `note` or `warn` or _&c._) on the diagnostic builder ... or are macros the way to go here? I think this is out-of-scope for this PR.)